### PR TITLE
Add Gridded Observations portal

### PR DESCRIPTION
**Pull request by corviday**
Add a new portal for gridded observational / historical data. ~~This portal requires the user to designate an area for download using the map, which is an end-run around issue #37 . To get the entire spatial domain, you'd draw a rectangle around the whole thing.~~

[Docker for the new portal](http://docker-dev01.pcic.uvic.ca:30032/gridded_observations/map/)

### CODE REVIEW

File: doc/source/pdp.rst [hunk 1]
```diff
@@ -31,3 +31,6 @@ The pdp module
 
 .. automodule:: pdp.portals.vic_gen1
    :members:
+
```
Comment (jameshiebert) [inline doc/source/pdp.rst]: Way to keep the docs in sync!

File: pdp/static/js/pdp_controls.js [hunk 2]
```diff
@@ -308,6 +310,11 @@ RasterDownloadLink.prototype = {
 
     // Take the URL template and substitute each of the desired state variables
     getUrl: function () {
+        //if this portal requies the user to select an area, don't generate a URL
+        //unless they've made a selection.
+        if(this.requireArea && !this.areaSpecified) {
```
Comment (jameshiebert) [inline pdp/static/js/pdp_controls.js]: I'm not a big fan of this solution. I'd rather that we omit the i/j parameters by default (which is a valid OPeNDAP request) until the user selects a rectangle.

This map require some changes to the `RasterDownloadLink` in `pdp_controls.js` or simply initializing the `RasterDownloadLink` with empty coordinates.

File: pdp/portals/gridded_observations.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from pdp.portals import updateConfig, raster_conf
+
+ensemble_name = 'gridded-obs-met-data'
+url_base = 'gridded_observations'
+
+class GriddedObservationsEnsembleLister(EnsembleMemberLister):
+
+    def list_stuff(self, ensemble):
+        dataset_names = {
+            "ANUSPLIN_CDA_v2012.1": "ANUSPLIN",
```
Comment (jameshiebert) [inline pdp/portals/gridded_observations.py]: We'll have to wait to hear back from Hydro, but I'd like to omit ANUSPLIN and SYMAP until #37 is fixed.

File: pdp/static/js/gridded_observations_app.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    catalogUrl = "../catalog/catalog.json";
+    catalog_request = $.ajax(catalogUrl, {dataType: "json"});
+
+    capabilities_request = getNCWMSLayerCapabilities(ncwmsLayer);
+
+    document.getElementById("pdp-controls").appendChild(getObsControls(pdp.ensemble_name));
+    document.getElementById("pdp-controls").appendChild(getRasterDownloadOptions(true));
+
+    // Data Download Link
+    dlLink = new RasterDownloadLink($('#download-timeseries'), ncwmsLayer, undefined, 'nc', 'sm', '0:54787', '0:163', '0:215', true);
```
Comment (jameshiebert) [inline pdp/static/js/gridded_observations_app.js]: Here, you may want to change '0:163' to '' and '0:215' to ''.

**Comment by corviday** [comment]
Here is what I hope is the last test docker on this project. It is built on [this branch of pydap-pdp](https://github.com/pacificclimate/pydap-pdp/pull/2), and initializes spatial indices on the Gridded Observations and Vic Gen1 portals to `[][]`:

http://docker-dev01.pcic.uvic.ca:30032/gridded_observations/map/

It all seems to be working, as far as I can tell! 

It's not a perfect systems / integration test: it's using an ncWMS2 instance running on docker-dev01 (that only contains the Gridded Observation files) instead of the ncWMS-PCIC instance on pizza.

**Pull request merged**